### DISCUSSION TITLE
Need guidance on Converting an Custom Trigger from In-Process to Isolated Mode

### DISCUSSION THREAD

**Discussion by ssinaikhandeparker** [Azure/azure-functions-dotnet-worker#2734] [Feedback]
We are currently working on migrating our function apps to Isolated Mode from the in-process mode. As of now we have successfully done the migration for function apps that use standard function triggers like timer trigger, servicebus trigger etc. 

However we are facing issues migrating functions that use a custom triggers. We have a few functions which are subscribing to an external source for polling and fetching messages messages. To handle this behavior we wrote a custom trigger using in-process mode. But now since we are migrating to isolated mode we would like to convert this custom trigger to the isolated mode. 

The Custom Trigger I am referring above is named as "UMTrigger". It is used to subscribe to an external source. This UMTrigger was built for the in-process mode using the Microsoft.Azure.WebJobs library. Now to convert it to work in isolated mode we are trying to replace relevant Microsoft.Azure.WebJobs to Microsoft.Azure.Functions.Worker. 

We seek some assistance on this topic which would suggest the relevant Worker Process libraries that we need replace or some guidance material on how to build a custom trigger in isolated mode. We have tried to look for any documentation which can guide us. But there is not enough material on this subject.

**Comment by arkiaman** [comment]
@ssinaikhandeparker maybe [that](https://github.com/arkiaconsulting/akc-azure-functions-worker-extensions-activemq) will help. You'll find there a repo that contains the code of an ActiveMQ custom trigger (in-process/isolated)

**Reply by ssinaikhandeparker** [reply]
Thanks @arkiaconsulting for the response. It seems for making the custom trigger to work in isolated mode, the trigger attribute need to now implement TriggerBindingAttribute of Microsoft.Azure.Functions.Worker.Extensions.Abstractions. Also a local nuget package is created of this project. 

I have done both these things but it isnt working for me. Is there any other thing that I am missing?

**Reply by Kryptos-FR** [reply]
I also had the same issue that I mentioned in #2092 and #2118, and with a small example in https://github.com/Kryptos-FR/azure-functions-dotnet-worker-2092.

**Comment by OzBob** [comment]
Do custom bindings need to be built into a seperate project and nuget!?
https://github.com/Azure/azure-functions-dotnet-worker/issues/2092
partly blogged about here: @maartenba [2021 Custom bindings with Azure Functions .NET Isolated Worker](https://blog.maartenballiauw.be/post/2021/06/01/custom-bindings-with-azure-functions-dotnet-isolated-worker.html)
